src/Categories/Bicategory/Construction/Bimodules/Tensorproduct.agda

```diff
@@ -0,0 +1,729 @@
+        actionʳ ∘ᵥ T₃ ▷ Coequalizer.arr F₂⊗F₁ ∘ᵥ T₃ ▷ actionʳ₂◁F₁ ≈⟨ refl⟩∘⟨ ∘ᵥ-distr-▷ ⟩
+        actionʳ ∘ᵥ T₃ ▷ (Coequalizer.arr F₂⊗F₁ ∘ᵥ actionʳ₂◁F₁) ≈⟨ refl⟩∘⟨ ▷-resp-≈ actionʳSq ⟩
+        actionʳ ∘ᵥ T₃ ▷ (actionʳ ∘ᵥ T₃ ▷ Coequalizer.arr F₂⊗F₁) ≈⟨ refl⟩∘⟨ ⟺ ∘ᵥ-distr-▷ ⟩
+        actionʳ ∘ᵥ T₃ ▷ actionʳ ∘ᵥ Coequalizer.arr T₃∘[T₃∘F]Coequalizer ≈⟨ sym-assoc₂ ⟩
+        (actionʳ ∘ᵥ T₃ ▷ actionʳ) ∘ᵥ Coequalizer.arr T₃∘[T₃∘F]Coequalizer ∎
+        where
+          open hom.HomReasoning
+
+      assoc-actionʳ : actionʳ ∘ᵥ μ₃ ◁ F ∘ᵥ associator.to ≈ actionʳ ∘ᵥ T₃ ▷ actionʳ
+      assoc-actionʳ = Coequalizer⇒Epi (hom C₁ C₃) T₃∘[T₃∘F]Coequalizer
+                                      (actionʳ ∘ᵥ μ₃ ◁ F ∘ᵥ associator.to)
+                                      (actionʳ ∘ᵥ T₃ ▷ actionʳ)
+                                      assoc-actionʳ∘arr
+
+      assoc-actionʳ-var : (actionʳ ∘ᵥ μ₃ ◁ F) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ T₃ ▷ actionʳ
+      assoc-actionʳ-var = begin
+        (actionʳ ∘ᵥ μ₃ ◁ F) ∘ᵥ associator.to ≈⟨ assoc₂ ⟩
+        actionʳ ∘ᵥ μ₃ ◁ F ∘ᵥ associator.to ≈⟨ assoc-actionʳ ⟩
+        actionʳ ∘ᵥ T₃ ▷ actionʳ ∎
+        where
+          open hom.HomReasoning
+
+      sym-assoc-actionʳ : actionʳ ∘ᵥ T₃ ▷ actionʳ ∘ᵥ associator.from ≈ actionʳ ∘ᵥ μ₃ ◁ F
+      sym-assoc-actionʳ = begin
+        actionʳ ∘ᵥ T₃ ▷ actionʳ ∘ᵥ associator.from ≈⟨ sym-assoc₂ ⟩
+        (actionʳ ∘ᵥ T₃ ▷ actionʳ) ∘ᵥ associator.from ≈⟨ ⟺ (switch-tofromʳ associator assoc-actionʳ-var) ⟩
+        actionʳ ∘ᵥ μ₃ ◁ F ∎
+        where
+          open hom.HomReasoning
+          open import Categories.Morphism.Reasoning.Iso (hom C₁ C₃)
+    -- end abstract --
+
+  module Identity where
+    open Left-Action using (actionˡ; actionˡSq; F₂▷actionˡ₁; F∘T₁Coequalizer)
+    open Right-Action using (actionʳ; actionʳSq; actionʳ₂◁F₁; T₃∘FCoequalizer)
+
+    abstract
+      identityˡ-triangle : F₂▷actionˡ₁ ∘ᵥ (F₂ ∘₁ F₁) ▷ η₁ ∘ᵥ unitorʳ.to ≈ id₂
+      identityˡ-triangle = begin
+        F₂▷actionˡ₁ ∘ᵥ (F₂ ∘₁ F₁) ▷ η₁ ∘ᵥ unitorʳ.to ≈⟨ assoc₂ ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ associator.from ∘ᵥ (F₂ ∘₁ F₁) ▷ η₁ ∘ᵥ unitorʳ.to ≈⟨ refl⟩∘⟨ sym-assoc₂ ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ (associator.from ∘ᵥ (F₂ ∘₁ F₁) ▷ η₁) ∘ᵥ unitorʳ.to ≈⟨ refl⟩∘⟨ α⇒-▷-∘₁ ⟩∘⟨refl ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ (F₂ ▷ F₁ ▷ η₁ ∘ᵥ associator.from) ∘ᵥ unitorʳ.to ≈⟨ refl⟩∘⟨ assoc₂ ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ F₂ ▷ F₁ ▷ η₁ ∘ᵥ associator.from ∘ᵥ unitorʳ.to ≈⟨ (refl⟩∘⟨ refl⟩∘⟨ ⟺ unitorʳ-coherence-var₂) ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ F₂ ▷ F₁ ▷ η₁ ∘ᵥ F₂ ▷ unitorʳ.to ≈⟨ refl⟩∘⟨ ∘ᵥ-distr-▷ ⟩
+        F₂ ▷ actionˡ₁ ∘ᵥ F₂ ▷ (F₁ ▷ η₁ ∘ᵥ unitorʳ.to) ≈⟨ ∘ᵥ-distr-▷ ⟩
+        F₂ ▷ (actionˡ₁ ∘ᵥ F₁ ▷ η₁ ∘ᵥ unitorʳ.to) ≈⟨ ▷-resp-≈ identityˡ₁ ⟩
+        F₂ ▷ id₂ ≈⟨ ▷id₂ ⟩
+        id₂ ∎
+        where
+          open hom.HomReasoning
+
+      identityˡ∘arr : (actionˡ ∘ᵥ F ▷ η₁ ∘ᵥ unitorʳ.to) ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈ id₂ ∘ᵥ Coequalizer.arr F₂⊗F₁
+      identityˡ∘arr = begin
+        (actionˡ ∘ᵥ F ▷ η₁ ∘ᵥ unitorʳ.to) ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈⟨ assoc₂ ⟩
```

Comment:
looks like there is a lot of 'refocusing' steps that can often be dealt with with push/pull. Definitely worth trying, so that the 'heart' of the proof becomes easier to see.